tools/setup.ps1

```diff
@@ -0,0 +1,38 @@
+
+$ROOT = ( get-item $PSScriptRoot ).Directory.parent.parent.FullName
+
+# Stop when error occurs
+$ErrorActionPreference = "Stop"
+
+$ChocoInstalled = $false
+
+function local:Status([String] $Message){
+  Write-Host $Message -ForegroundColor DarkMagenta
+}
+
+
+Status("1. Install nvm")
+if (Get-Command choco.exe -ErrorAction SilentlyContinue) {
+  $ChocoInstalled = $true
+  choco install nvm
+} else {
+  Install-Module -Name nvm
+}
+
+Status("1.1 Ensure nvm is in your current process")
+if($ChocoInstalled) {
+  refreshenv
+} else {
+  $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path","User")
+}
+
+Status("2. Install Node")
+CD $ROOT
+nvm install
+nvm use
+
+Status("3. Install dependencies")
+CD $ROOT
+npm ci
+
+
```

Comment:
I just noticed we're not running an initial `./tools/sigh` (Windows equivalent) here, or likely in the Linux version. Maybe we should? But we could keep setup separate. If we add it maybe we'd then need to remove extranous `sigh` run from our CI configs.

Reply:
I thought this happens by default when we `npm install` (or `npm ci` in this case)?

Reply:
A TODO that I've punted: printing suggested commands, like the original setup script. I wanted to open the PR to test that it works in appveyor.